### ISSUE TITLE
background class show nothing in confusion matrix

### ISSUE THREAD

**Issue by yueguangjiansheng** [open-mmlab/mmdetection#11361]
Thanks for your error report and we appreciate it a lot.

**Checklist**

1. I have searched related issues but cannot get the expected help.
2. I have read the [FAQ documentation](https://mmdetection.readthedocs.io/en/latest/faq.html) but cannot get the expected help.
3. The bug has not been fixed in the latest version.

**Describe the bug**
When I put my own data sets into mmdet-master, there was a blank line always in background class. 
AND a warning came out:tools/analysis_tools/confusion_matrix.py:184: RuntimeWarning: invalid value encountered in divide
  confusion_matrix.astype(np.float32) / per_label_sums * 100.  After I add the code 'np.seterr(divide='ignore', invalid='ignore')', the issue still exist. I guess that the background class did not go into ground true label, it only go into prediction label. so that per_label_sums =0, and the warning came out.
![confusion_matrix](https://github.com/open-mmlab/mmdetection/assets/94508529/dbe097b8-86a9-4d69-8029-e3a744e070b3)

**Reproduction**

1. What command or script did you run?
mmdetection-master\configs\deformable_detr\deformable_detr_r50_16x2_50e_coco.py
mmdetection-master\configs\faster_rcnn\faster_rcnn_r50_caffe_c4_1x_coco.py
mmdetection-master\tools\train.py
mmdetection-master\tools\test.py
mmdetection-master\tools\analysis_tools\confusion_matrix.py

2. Did you make any modifications on the code or config? Did you understand what you have modified?
there is no modification in code. I download and loading the right pre train model from your provided models.
3. What dataset did you use?
I used my own dataset, it consists of pictures of 13 varieties of pecan. I have labelled carefully and checked the datasets for many times.
Sorry about I cant share it now. 

**Environment**

1. sys.platform: win32
Python: 3.8.17 (default, Jul  5 2023, 20:35:33) [MSC v.1916 64 bit (AMD64)]
CUDA available: True
GPU 0: NVIDIA GeForce RTX 3070
CUDA_HOME: C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v11.3
NVCC: Not Available
GCC: n/a
PyTorch: 1.10.0+cu113
PyTorch compiling details: PyTorch built with:
  - C++ Version: 199711
  - MSVC 192829337
  - Intel(R) Math Kernel Library Version 2020.0.2 Product Build 20200624 for Intel(R) 64 architecture applications
  - Intel(R) MKL-DNN v2.2.3 (Git Hash 7336ca9f055cf1bfa13efb658fe15dc9b41f0740)
  - OpenMP 2019
  - LAPACK is enabled (usually provided by MKL)
  - CPU capability usage: AVX2
  - CUDA Runtime 11.3
  - NVCC architecture flags: -gencode;arch=compute_37,code=sm_37;-gencode;arch=compute_50,code=sm_50;-gencode;arch=compute_60,code=sm_60;-gencode;arch=compute_61,code=sm_61;-gencode;arch=compute_70,code=sm_70;-gencode;arch=compute_75,code=sm_75;-gencode;arch=compute_80,code=sm_80;-gencode;arch=compute_86,code=sm_86;-gencode;arch=compute_37,code=compute_37
  - CuDNN 8.2
  - Magma 2.5.4
  - Build settings: BLAS_INFO=mkl, BUILD_TYPE=Release, CUDA_VERSION=11.3, CUDNN_VERSION=8.2.0, CXX_COMPILER=C:/w/b/windows/tmp_bin/sccache-cl.exe, CXX_FLAGS=/DWIN32 /D_WINDOWS /GR /EHsc /w /bigobj -DUSE_PTHREADPOOL -openmp:experimental -IC:/w/b/windows/mkl/include -DNDEBUG -DUSE_KINETO -DLIBKINETO_NOCUPTI -DUSE_FBGEMM -DUSE_XNNPACK -DSYMBOLICATE_MOBILE_DEBUG_HANDLE -DEDGE_PROFILER_USE_KINETO, LAPACK_INFO=mkl, PERF_WITH_AVX=1, PERF_WITH_AVX2=1, PERF_WITH_AVX512=1, TORCH_VERSION=1.10.0, USE_CUDA=ON, USE_CUDNN=ON, USE_EXCEPTION_PTR=1, USE_GFLAGS=OFF, USE_GLOG=OFF, USE_MKL=ON, USE_MKLDNN=ON, USE_MPI=OFF, USE_NCCL=OFF, USE_NNPACK=OFF, USE_OPENMP=ON, 

TorchVision: 0.11.0+cu113
OpenCV: 4.8.0
MMCV: 1.4.2
MMCV Compiler: MSVC 192930137
MMCV CUDA Compiler: 11.3
MMDetection: 2.23.0+
2.无

**Error traceback**
C:\ProgramData\Anaconda3\envs\mmclassification-master\python.exe D:/CBY/mmdetection-master/tools/analysis_tools/confusion_matrix.py D:\\CBY\\mmdetection-master\\configs\\faster_rcnn\\my_faster_rcnn_r50_caffe_c4_1x_coco.py D:\\CBY\\mmdetection-master\\tools\\work_dirs\\faster_rcnn_r50_caffe_c4_1x_coco\\test.pkl D:\\CBY\\mmdetection-master\\tools\\work_dirs\\faster_rcnn_r50_caffe_c4_1x_coco\\ --show
loading annotations into memory...
Done (t=0.01s)
creating index...
index created!
[>>>>>>>>>>>>>>>>>>>>>>>>>>>>] 560/560, 2995.5 task/s, elapsed: 0s, ETA:     0sD:/CBY/mmdetection-master/tools/analysis_tools/confusion_matrix.py:184: RuntimeWarning: invalid value encountered in divide
  confusion_matrix.astype(np.float32) / per_label_sums * 100

**Bug fix**
I am curious about the cause of this problem. If possible, please modify the code. If you need to view my dataset, please email me and I can use the Sunflower software to remotely share the screen. chenbingyu@stu.zafu.edu.cn
